### DISCUSSION TITLE
Declarative DI

### DISCUSSION THREAD

**Discussion by jamalsalman** [dotnet/aspnetcore#51933] [Ideas] [answered]
Hey,
Sometimes, especially when building the smaller projects, I secretly wish that aspnetcore supports attribute based DI like the one here 
[https://github.com/mrahhal/MR.AttributeDI](https://github.com/mrahhal/MR.AttributeDI).
I'm curious to see what the community thinks about that, and why it hasn't been considered.

**Comment by gfoidl** [comment]
The link provided doesn't work. @jamalsalman please fix that.

**Reply by jamalsalman** [reply]
Sorry about that, should be good now

**Comment by martincostello** [comment, marked as answer]
The .NET dependency injection support lives in the [dotnet/runtime repository](https://github.com/dotnet/runtime). Adding support for attributes has previously been discussed in this issue: https://github.com/dotnet/runtime/issues/36749
